main.go

```diff
@@ -123,3 +145,12 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func getSecret(secretName string) []byte {
+	secret, err := ioutil.ReadFile(dbSecretBasePath + secretName)
+	if err != nil {
+		setupLog.Error(err, "error reading secret")
+		os.Exit(1)
+	}
```

Comment:
Let's update the function signature here and return an explicit error and letting the call site handle the error.